text/0000-match-ergonomics.md

```diff
@@ -3,11 +3,15 @@
 - RFC PR: (leave this empty)
 - Rust Issue: (leave this empty)
 
+I'm writing comments inline (not making changes), hopefully they'll stand out in the diff.
+
 # Summary
 [summary]: #summary
 
 Better ergonomics for pattern-matching on references.
 
+AIUI, this will only apply to match and similar expressions, not pattern matching in general?
```

Comment:
@nikomatsakis and @nagisa [thought it would be nice to have a consistent meaning to all patterns](https://github.com/rust-lang/rfcs/pull/1944#issuecomment-296170541). I tend to agree.